scipy/stats/_stats_py.py

```diff
@@ -100,13 +100,20 @@ def _contains_nan(a, nan_policy='propagate'):
         try:
             contains_nan = np.any(np.isnan(a))
         except TypeError:
-            # Don't know what to do. Fall back to omitting nan values and
-            # issue a warning.
-            contains_nan = False
-            nan_policy = 'omit'
-            warnings.warn("The input array could not be properly "
-                          "checked for nan values. nan values "
-                          "will be ignored.", RuntimeWarning)
+            try:
+                # This can happen when attempting to check nan with np.isnan
+                # for string array (e.g. as in the function `rankdata`).
+                a = np.array(a)
+                nan_str = np.array([np.nan]).astype(str)[0]
+                contains_nan = a[np.where(a.astype(str) == nan_str)].size != 0
```

Comment:
```suggestion
                contains_nan = np.any(a.astype(str) == nan_str)
```
More concise and seems more readable to me.